test/src/runner/testRunnerScreenshotsOutput.js

```diff
@@ -0,0 +1,313 @@
+const path = require('path');
+const fs = require('fs');
+const assert = require('assert');
+const common = require('../../common.js');
+const MockServer = require('../../lib/mockserver.js');
+const CommandGlobals = require('../../lib/globals/commands.js');
+const NightwatchClient = common.require('index.js');
+const rimraf = require('rimraf');
+
+describe('testRunnerScreenshotsOutput', function() {
+  
+  before(function(done) {
+    this.server = MockServer.init();
+    this.server.on('listening', () => {
+      if(fileExistsSync('screenshots/')){
+        rimraf.sync('screenshots/');
+      }
+      done();
+    });
+  });
+
+  afterEach(function(done) {
+    if(fileExistsSync('screenshots/')){
+      rimraf.sync('screenshots/');
+    }
+    done();
+  });
+
+  after(function(done) {
+    CommandGlobals.afterEach.call(this, function() {
+      if(fileExistsSync('screenshots/')){
+        rimraf.sync('screenshots/');
+      }
+      done();
+    });
+  });
+
+  beforeEach(function() {
+    process.removeAllListeners('exit');
+    process.removeAllListeners('uncaughtException');
+    process.removeAllListeners('unhandledRejection');
+  });
+
+  it('takes screenshot on each test failure', function () {
+
+
+    let testsPath = [
+      path.join(__dirname, '../../sampletests/withfailures')
+    ];
+
+    let settings = {
+      skip_testcases_on_fail: false,
+      selenium: {
+        port: 10195,
+        version2: true,
+        start_process: true
+      },
+      output_folder: 'output',
+      silent: true,
+      globals: {
+        waitForConditionPollInterval: 5,
+        waitForConditionTimeout: 5,
+        retryAssertionTimeout: 1,
+        reporter: function () {
+        }
+      },
+      output: false,
+      screenshots: {
+        enabled: true,
+        on_failure: true,
+        on_error: true,
+        path: 'screenshots'
+      }
+    };
+
+    MockServer.addMock({
+      url : '/wd/hub/session/1352110219202/screenshot',
+      method:'GET',
+      response : JSON.stringify({
+        sessionId: '1352110219202',
+        status:0,
+        value:'screendata'
+      })
+    }, true);
+
+    return NightwatchClient.runTests(testsPath, settings)
+      .then(_ => {
+        return readDirPromise('screenshots/sample');
+      })
+      .then(files => {
+        assert.ok(files);
```

Comment:
- `.ok()` should only be used when checking a boolean; use `Array.isArray(files)` here
- `deepStrictEqual` is only used when asserting objects; use `strictEqual` 
- use `assert.ok(val.endsWith('.png'))`